CHANGES.md

```diff
@@ -1,6 +1,490 @@
+* deprecated default `true` value for `locking` arg in `openReader|Writer()`  
+  (see https://chapel-lang.org/docs/2.0/modules/standard/IO.html#IO.OpenReaderLockingDefault  
+   and https://chapel-lang.org/docs/2.0/modules/standard/IO.html#IO.OpenWriterLockingDefault)
+* deprecated default value for `locking` argument in `file.reader|writer()`  
+  (see https://chapel-lang.org/docs/2.0/modules/standard/IO.html#IO.file.reader  
+   and https://chapel-lang.org/docs/2.0/modules/standard/IO.html#IO.file.writer)
+* unstabilized `imag` overloads of trigonometric/hyperbolic 'Math' functions
+* removed the unstable `imag` overloads of the `fma()` procedure
+* removed the remaining deprecated routines/constants from the 'Math' module  
+  (e.g., `carg()`, `conjg()`, `divceil()`, `log2_e`, `half_pi`, etc.)
+* removed deprecated 'IO' support for the `readThis()`/`writeThis()` methods
+* removed the deprecated 'IO' configs that control `region` argument behavior  
+  (e.g. `useNewSeekRegionBounds`, `useNewOpenReaderRegionBounds`, etc.)
+* removed all previously deprecated `BigInteger` functions
+* removed the deprecated 'BigInteger' config `bigintInitThrows`
+* removed some of the previously deprecated `Time.date[time]` methods
+* removed the previously deprecated `Time.getCurrentTime()` function
+* removed some of the previously deprecated `Time` types
+  (e.g., `datetime`, `timedelta`, `Timer`, `TZInfo`, `TimeUnits`, etc.)
+* removed all previously deprecated `FileSystem` functions
+* removed the previously deprecated `BitOps.popcount()` function
+* removed `iokind` from the 'IO' module
+* removed `iostyle` from the 'IO' module
+* removed `file[Reader|Writer].binary` from the 'IO' module
+* removed `file[Reader|writer].writing` from the 'IO' module
+* removed all pascal-cased (de)serializers from the 'IO' module
+* removed `ioLiteral` and `ioNewLine` from the 'IO' module
+* removed `readWriterLiteral` and `readWriteNewline` from the 'IO' module
+* removed `readline` from the 'IO' module
+* removed `stringify` from the 'IO' module
+* removed `ioHintSet.noMmap` from the 'IO' module
+* removed the `filePathAbsolute` flag from the 'IO' module
+* removed the `%t`, `%jt`, and `%ht` format specifiers from 'FormattedIO'
+* removed the binary (`|`) format specifier from 'FormattedIO'
+* removed the old left-justification (`-`) format specifier from 'FormattedIO'
+* removed the deprecated support for casting any expression to a string
+
+GPU Computing
+-------------
+* replaced `setBlockSize` with a new `@gpu.blockSize` loop attribute
+* added support for `@assertOnGpu` and `@gpu.blockSize` on variable decls
+* improved GPU sorting routines w.r.t. performance and supported types  
+  (see https://chapel-lang.org/docs/2.0/modules/standard/GPU.html#GPU.gpuSort)
+* adjusted the compiler to print traces of call chains that thwart GPUization
+* enabled using ROCm's LLVM using `CHPL_LLVM=system` and `CHPL_GPU=amd`
+  (see TODO)
```

Comment:
The documentation for this will be in:

https://chapel-lang.org/docs/2.0/technotes/gpu.html#requirements